lib/file.js

```diff
@@ -104,8 +115,74 @@ FileCache.prototype.getCachePath = function (key) {
   return path.resolve(path.join(folderPath, key + this.extension))
 }
 
+
+
+/**
+ * enableCacheCleansing - sets up and manages the timeout for cache cleansing
+ *
+ * @returns {void}
+ */
+FileCache.prototype.enableCacheCleansing = function enableCacheCleansing() {
+  var timeoutTrigger = () => {
+    this.cleanseTimeout = setTimeout(() => {
+      this.cleanseEnabled && this.cleanseCache(this.directory, timeoutTrigger)
+    }, this.cleanseInterval * 1000)
+  }
+
+  timeoutTrigger()
+}
+
 /**
+ * disableCacheCleansing - stops and cleans up the timeout for cache cleansing
  *
+ * @returns {void}
+ */
+FileCache.prototype.disableCacheCleansing = function enableCacheCleansing() {
+  this.cleanseEnabled = false
+  if (this.cleanseTimeout) {
+    clearTimeout(this.cleanseTimeout)
+  }
+}
+
+/**
+ * cleanseCache - trawls cache files and removes any that have expired
+ *
+ * @param {String} directory path to cleanse
+ * @param {callback} done
+ * @returns {void}
+ */
+FileCache.prototype.cleanseCache = function cleanseCache(dir, done) {
+  recursive(dir, (recursiveErr, files) => {
+    if (recursiveErr) {
+      throw recursiveErr
+    }
+
+    var count = files.length
+    var idx = 0
+
+    files.forEach((file) => {
+      fs.stat(file, (statErr, stats) => {
+        if (statErr) {
+          throw statErr
+        }
+
+        var lastModified = stats && stats.mtime && stats.mtime.valueOf()
+
+        if (this.ttl && lastModified && (Date.now() - lastModified) / 1000 > this.ttl) {
```

Comment:
do we need to test for `this.ttl` if it's defaulting to 3600 in the constructor?